### DISCUSSION TITLE
a RolePolicy to restrict access to the Role ( same for Permissions) not being honored.

### DISCUSSION THREAD

**Discussion by ddoddsr** [spatie/laravel-permission#2461] [Q&A] [answered]
I made a RolePolicy with a with viewAny return false
but the policy is not honored.  ( All users can view Roles )
What am I missing??

```
<?php

namespace App\Policies;

use App\Models\SpatiePermissionModels\Role;
use App\Models\User;
use Illuminate\Auth\Access\Response;

class RolePolicy
{
    public function viewAny(User $user): bool
    {
        return false;
    }
}

```

**Comment by drbyte** [comment, marked as answer]
I think we'd need to see more of your application code. Perhaps even create a simple demo application that shows what you're trying to do specifically in this regard.

Chances are it's related to your policy not being loaded/detected, and/or your controller or routes/middleware code.

**Comment by ddoddsr** [comment]
Yes it was due to the policy not being loaded correctly.
